api/v1/checks.go

```diff
@@ -628,13 +686,86 @@ func (c SMBConnection) GetPort() int {
 	return 445
 }
 
+func (c *SMBConnection) PopulateFromConnection(ctx context.Context, db *gorm.DB, k8sClient kubernetes.Interface, namespace string) (found bool, err error) {
+	if c.ConnectionName == "" {
+		return false, nil
+	}
+
+	connection, err := duty.HydratedConnectionByURL(ctx, db, k8sClient, namespace, c.ConnectionName)
+	if err != nil {
+		return false, err
+	}
+
+	if connection == nil {
+		return false, nil
+	}
+
+	c.Domain = connection.URL
```

Comment:
Domain is something that comes from the username e.g. johndoe@acme.corp and the domain is acme.corp

Reply:
@moshloop I think we should probably do it while we're saving the connection.

Reply:
ok, then pull domain from the properties